src/screens/WhotoFollow/index.js

```diff
@@ -173,65 +173,65 @@ const WhotoFollow = () => {
-          setToken(res.token);
-          try {
-            const userId = await JwtDecode(res.token).user_id;
-            const anonymousUserId = await JwtDecode(res.anonymousToken).user_id;
-            setProfileId({
-              anonProfileId: anonymousUserId,
-              signedProfileId: userId,
-              token: res.token,
-              anonymousToken: res.anonymousToken
+      registerUser(data)
+        .then(async (res) => {
+          setFetchRegister(false);
+          if (res.code === 200) {
+            TokenStorage.set(res);
+            setToken(res.token);
+            try {
+              const userId = await JwtDecode(res.token).user_id;
+              const anonymousUserId = await JwtDecode(res.anonymousToken).user_id;
+              setProfileId({
+                anonProfileId: anonymousUserId,
+                signedProfileId: userId,
+                token: res.token,
+                anonymousToken: res.anonymousToken
+              });
+            } catch (e) {
+              crashlytics().recordError(new Error(e));
+            }
+            showMessage({
+              message: 'Welcome to Better Social',
+              type: 'success',
+              backgroundColor: colors.holytosca
+            });
+            setTimeout(() => {
+              create();
+              setImage(null, usersDispatch);
+              setInitialValue({id: res.token});
+            }, 2000);
+          } else {
+            crashlytics().recordError(new Error(res));
+            showMessage({
+              message: 'Cannot connect to server, please try again later',
+              type: 'danger',
+              backgroundColor: colors.red
             });
-          } catch (e) {
-            crashlytics().recordError(new Error(e));
           }
-          showMessage({
-            message: 'Welcome to Better Social',
-            type: 'success',
-            backgroundColor: colors.holytosca
-          });
-          setTimeout(() => {
-            create();
-            setImage(null, usersDispatch);
-            setInitialValue({id: res.token});
-          }, 2000);
-        } else {
-          crashlytics().recordError(new Error(res));
+        })
+        .catch((error) => {
+          crashlytics().recordError(new Error(error.response));
+          setFetchRegister(false);
           showMessage({
             message: 'Cannot connect to server, please try again later',
             type: 'danger',
             backgroundColor: colors.red
           });
-        }
-      })
-      .catch((error) => {
-        crashlytics().recordError(new Error(error.response));
-        setFetchRegister(false);
-        showMessage({
-          message: 'Cannot connect to server, please try again later',
-          type: 'danger',
-          backgroundColor: colors.red
         });
-      });
+    } catch (e) {
+      console.error('Error during registration:', e);
+      // Handle the error appropriately here
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling in the `registerUser` function has been improved by wrapping the entire function in a try-catch block. However, there is still room for improvement. The catch block at lines 231-234 only logs the error to the console and does not provide any user feedback or handle the error in a way that would allow the application to recover gracefully. Consider adding user feedback and error recovery mechanisms similar to those used in the `.catch()` block of the `registerUser` promise.

```diff
-      console.error('Error during registration:', e);
-      // Handle the error appropriately here
+      crashlytics().recordError(new Error(e));
+      setFetchRegister(false);
+      showMessage({
+        message: 'An error occurred during registration, please try again later',
+        type: 'danger',
+        backgroundColor: colors.red
+      });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
Fixed the feedback